### PR TITLE
Create, Collect, and Publish JUnit Reports from Static Type Checking

### PR DESCRIPTION
**Pull request by Paebbels**
# New Features
* Create junit summary files from static type checking (`mypy`).
* Collect the junit file as artifact.
* Publish this artifact like other test summary reports.

# Changes
* Renamed `report` and `artifact` in the StaticTyping job template to `html_report` and `html_artifact`.
* Added new `junit_report` and `junit_artifact` parameters.
* Adjusted Parameters job
* Adjusted Example pipeline
* Adjusted artifact cleanup.

# Bug Fixes
*None*

---------------
Related PRs:
* #40

### CODE REVIEW

File: .github/workflows/CoverageCollection.yml [hunk 1]
```diff
@@ -35,10 +35,15 @@ on:
         required: false
         default: '-r tests/requirements.txt'
         type: string
+      test_directory:
+        description: 'Path to the directory containing tests (test working directory).'
```
Comment (umarcor) [inline .github/workflows/CoverageCollection.yml:39]: The working directory does not need to be the one containing the tests. Therefore, I would use `workdir` for this input variable.

File: .github/workflows/CoverageCollection.yml [hunk 2]
```diff
@@ -117,15 +122,19 @@ jobs:
       - name: Collect coverage
         continue-on-error: true
         run: |
-          [ 'x${{ inputs.coverage_config }}' != 'x' ] && PYCOV_ARGS='--cov-config=${{ inputs.coverage_config }}' || unset PYCOV_ARGS
-          python -m pytest -rA --cov=. $PYCOV_ARGS ${{ inputs.unittest_directory }} --color=yes
+          RELDIR="$(realpath --relative-to=${{ inputs.test_directory }} .)"
```
Comment (umarcor) [inline .github/workflows/CoverageCollection.yml:125]: Why do we need the coverage config to be relative to the workdir? We can get the absolute location (through `pwd`), before `cd ${{ inputs.testdirectory }}`.

File: .github/workflows/StaticTypeCheck.yml [hunk 3]
```diff
@@ -35,19 +35,29 @@ on:
...
-        description: 'Commands to run the static type checks.'
-        required: true
+      junit_report:
+        description: 'junit file to upload as an artifact.'
+        required: false
+        default: 'StaticTypingSummary.xml'
         type: string
-      artifact:
+      html_artifact:
         description: 'Name of the typing artifact.'
```
Comment (umarcor) [inline .github/workflows/StaticTypeCheck.yml:53]: We should make this description different from the one of "junit_artifact".

File: .github/workflows/UnitTesting.yml [hunk 4]
```diff
@@ -48,10 +48,15 @@ on:
         required: false
         default: ''
         type: string
+      test_directory:
```
Comment (umarcor) [inline .github/workflows/UnitTesting.yml:51]: See comment about the name of this input, above.

File: .github/workflows/StaticTypeCheck.yml (hunk 3 above)
Comment (Paebbels) [reply .github/workflows/StaticTypeCheck.yml:53]: Fixed in 301584a670de754337a6f1c38e07f68cde79a656 and integrated into #51.

File: .github/workflows/CoverageCollection.yml (hunk 2 above)
Comment (Paebbels) [reply .github/workflows/CoverageCollection.yml]: Both solution require a similar amount of extra scripting / variables. Coverage is not executed on macOS, so no problems with `realpath`.

**Pull request merged**